api/src/main/java/ai/djl/ndarray/BaseNDManager.java

```diff
@@ -206,7 +209,17 @@ public synchronized void attach(String resourceId, AutoCloseable resource) {
 
     /** {@inheritDoc} */
     @Override
-    public synchronized void detach(String resourceId) {
+    public void tempAttachInternal(
```

Comment:
what makes difference on tempAttachInternel comparing to the method attachInternal?